### DISCUSSION TITLE
"quiesce" alerts for a camera for a while after 1st alert sent

### DISCUSSION THREAD

**Discussion by obermotz** [0x2142/frigate-notify#70] [Ideas]
Just installed frigate-notify, works splendidly. I would like to propose a new feature though: for my use case it would be great if there was a way to not send new alerts for a camera for a configurable length of time after sending out the 1st one.

Thanks in advance for considering this.

**Comment by freefd** [comment]
Dear @obermotz,

Today, the frigate-notify lifecycle of notifications works as follows (simplified):
![image](https://github.com/0x2142/frigate-notify/assets/133116/0c181772-9faf-4f35-9e12-e31541d8c046)

For the MQTT-based approach, the decision on when to send a new notification is completely controlled by Frigate using the Event.Type field: frigate-notify works only with the “New” and “Update” + Zone Change modifiers.
This way, you will only receive one notification for each event in each Zone (if configured).

For the WebAPI approach, there is no additional logic in the polling: [any new appeared event](https://github.com/0x2142/frigate-notify/blob/main/events/api.go#L31-L32) will be processed by the frigate-notify.

> ... for my use case ...

Hence, could you please explain your use case a bit more in detail?

**Comment by obermotz** [comment]
Dear @freefd ,

i have an MQTT based setup and my use case is something like this: I have a building watched by one of my cameras. I would like to get a notification when a person enters the building, but the very same person will then enter/exit the building several times in just a couple of minutes and I am not interested in the alerts generated by these movements. So I would like to receive the 1st alert (so I know someone is there), then suppress alerts / ignore MQTT events associated with that particular camera for like 5 minutes.

**Comment by freefd** [comment]
Dear @obermotz,

If your case could be depicted as the following:
![image](https://github.com/0x2142/frigate-notify/assets/133116/90bd8112-aad3-48be-969c-43ed1dae6825)


You may try to configure zones in the Frigate:
![image](https://github.com/0x2142/frigate-notify/assets/133116/a644157f-13ac-465c-89e8-d876a737a0a1)

And later ignore Zone2 by frigate-notify configuration: https://frigate-notify.0x2142.com/config/#zones

Also, you can try Frigate-only configuration with `required_zones` feature: https://docs.frigate.video/guides/parked_cars/

**Comment by 0x2142** [comment]
Hmm - That's an interesting use case & thanks for sharing the info! Right now frigate-notify is based entirely on the events as they come from Frigate & I'm not storing any kind of state info between events. So when each event comes in, frigate-notify has no knowledge of any events prior. So far I've been aiming to keep it this way, since it's much simpler to manage & I would prefer to avoid needing a database - but perhaps we'll need to adapt over time. Something like this might still be possible without too much overhead, but I'll need to try a few things.

That being said, I do wonder with the scenario above... That would likely be several events from Frigate if the person was in & out of the building over a few minutes. If we mute the camera notifications for a few minutes, what if someone different entered during that time? Or are you only interested in being notified that _anyone_ is at the building?

**Reply by obermotz** [reply]
Or are you only interested in being notified that anyone is at the building? - yep, this is it

**Comment by MaxKuh** [comment]
First of all thanks also from my side. Frigate-Notify is awesome and working flawlessly. 
I want to add my thoughts to this thread, since I was searching for the same functionality.

Since the new frigate version events are grouped together, i.e. two detected objects lead to only one video/ alarm that is captured. Nevertheless frigate-notify sends out several messages (since every new detected object or a new zone will lead to a new alarm). 

From my usecase it would be enough to receive only one notification, according to the one event in frigate (v0.14).

Right now I receive several alarms from frigate-notify (usecase camera in driveway: car is detected in driveway (alarm), people exit car and are detected (alarm), people enter new zone closer to the house and are detected again (alarm)). So that is at least 3 alarms from frigate-notify but only one event in frigate itself. For me one alarm would be sufficient, since I know then that something is going on and I can check if I don't know the car.

My initial idea was a minimum time between two alarms (cooldown phase or however you want to call it). In my case I receive several notifications within a few seconds or even two events at the same time. 

Do you have a suggestion on how to deal with that?
Edit: I am using the MQTT approach.

**Reply by 0x2142** [reply]
While I'm still working for support on Frigate v0.14's new alerts/reviews - I went ahead and added a quick option for `notify_once`, where you would only ever get one notification per event from Frigate. Docs [here](https://frigate-notify.0x2142.com/latest/config/file/#general) on that, if you're interested

**Reply by MaxKuh** [reply]
Awesome, thanks. I still receive an alarm for every single person, but I guess that is caused by the single events that every person generates?!

**Comment by freefd** [comment]
Linking here the same behavior with the `Zoned` and `Unzoned` events when `unzoned` is set to `allow`.
https://github.com/0x2142/frigate-notify/issues/66

**Comment by freefd** [comment]
@MaxKuh,

> My initial idea was a minimum time between two alarms (cooldown phase or however you want to call it). In my case I receive several notifications within a few seconds or even two events at the same time.

The cooldown phase may provide false negative results as from the Frigate point of view all those events are exposed for the DIFFERENT things: a car detection in `Zone A`, a person detection in `Zone A`, and the SAME person detection in `Zone B` afterwards.

The more intelligence solution is required here, and I tried to explain it in the https://github.com/0x2142/frigate-notify/issues/66 issue:
> Some algorithm for calculating the distance between polygons (geo libs) may be needed here: smaller numbers may mean that the object is simply moving from `Zone A` to `Zone B`.

Moreover, each event contains coordinates as region, event samples were mentioned in https://github.com/0x2142/frigate-notify/issues/64.
